src/webots/gui/WbX3dStreamingServer.cpp

```diff
@@ -64,10 +65,11 @@ void WbX3dStreamingServer::create(int port) {
 
 void WbX3dStreamingServer::sendTcpRequestReply(const QString &url, const QString &etag, const QString &host,
                                                QTcpSocket *socket) {
-  if (!mX3dWorldTextures.contains(url))
-    WbTcpServer::sendTcpRequestReply(url, etag, host, socket);
+  QFileInfo file(WbProject::current()->dir().absolutePath() + "/" + url);
+  if (file.exists())
+    socket->write(WbHttpReply::forgeFileReply(file.absoluteFilePath(), etag, host, url));
```

Comment:
Did you check that we do not lost performance here?

Reply:
How so? The only difference compared to before is that we check if the file exists rather than checking if the map contains it